### PR TITLE
Initial autoupdate implementation

### LINKED ISSUE #126: Autoupdate GA

### PR DESCRIPTION
**Pull request by main--**
Needs more testing.

Fixes #126.

### CODE REVIEW

**Comment by oberien** [comment]
Is it possible to attach the cd drive dynamically with `device_add` only if the GA is outdated, and remove it again when the GA is updated? I think it feels weird to have the GA.iso attached at all times.

Otherwise lgtm

**Comment by main--** [comment]
@oberien That's what I was planning to do initially. However, there's no straightforward way to tell if/when the device is going to be ready. Plus, implementing the required QMP commands is somewhat weird (see #131).

Permanently attaching the GA ISO is definitely not perfect but not that big of a deal either. It's there, whatever, just ignore it.

Hypervisors have traditionally addressed this by creating an optical drive at boot time and allowing you to insert/remove ISO images at runtime. This does not work here however as the corresponding qemu APIs are still unstable/experimental.

**Comment by oberien** [comment]
You should create a new issue then just to keep it in mind. (Also don't forget to update the reference in #131 ).

If it wasn't for "Needs more testing.", I'd approve this PR.

File: windows-gaming/driver/src/controller.rs [hunk 1]
```diff
@@ -127,9 +128,25 @@ impl Controller {
...
 
-    pub fn ga_hello(&mut self) -> bool {
+    pub fn ga_hello(&mut self, version: String) -> bool {
         sd_notify::notify_systemd(true, "Ready");
 
+        // check for autoupdate
+        let latest = Version::parse(include_str!("../../../guest-agent/VfioService/VfioService/Resources/Version.txt")).unwrap();
+        match Version::parse(&version) {
+            Ok(version) => {
+                if version < latest {
```
Comment (moritzuehling) [inline windows-gaming/driver/src/controller.rs]: != instead of <

File: windows-gaming/driver/src/qemu.rs [hunk 2]
```diff
@@ -46,6 +46,9 @@ pub fn run(cfg: &Config, tmp: &Path, data: &Path, clientpipe_path: &Path, monito
         debug!("Samba started");
     }
 
+    let ga = data.join("windows-gaming-ga.iso");
```
Comment (moritzuehling) [inline windows-gaming/driver/src/qemu.rs]: naming: `gaIso` or something like that to easier see what this is. `ga` can have multiple meanings

**Review by moritzuehling — COMMENTED** [review]
Gernerally I'd probably prefer one of the two methods we discussed.

**Review by moritzuehling — APPROVED** [review]

**Review by oberien — DISMISSED** [review]

**Pull request merged**